x-pack/elastic-agent/pkg/agent/application/info/agent_id.go

```diff
@@ -43,11 +44,16 @@ func AgentConfigFile() string {
 	return filepath.Join(paths.Config(), defaultAgentConfigFile)
 }
 
-// AgentActionStoreFile is the file that will contains the action that can be replayed after restart.
+// AgentActionStoreFile is the file that contains the action that can be replayed after restart.
```

Comment:
Being that we already have an `action_store.yml` why not place the action token inside of this file? Why the need to seperate it into its own file?

Reply:
can change. was not sure what's the right/established pattern.

Reply:
is there any case where the `agent_store.yml` could get completely overwritten, thus loosing the marker?

Reply:
i think that would be the correct place

It should only be overwritten on a re-enroll, which is what should happen in the token case. The code paths already handle that, so its the best place for it.